From 98494cbc35117a83cff5e0c23d200243bdff4f43 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Tue, 12 Apr 2022 11:21:21 +0000
Subject: [PATCH 085/120] WIP: arm64: dts: meson: sm1-x96-air: add RTL8822CS
 WiFi and Bluetooth support

Add support for the RTL8822CS WiFi and BT module on the board.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../dts/amlogic/meson-sm1-x96-air-gbit.dts    | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts
index 7e1a74046ba5..9217db5bdcea 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-x96-air-gbit.dts
@@ -13,6 +13,10 @@ / {
 	compatible = "amediatech,x96-air-gbit", "amlogic,sm1";
 	model = "Shenzhen Amediatech Technology Co., Ltd X96 Air";
 
+	aliases {
+		serial2 = &uart_A;
+	};
+
 	sound {
 		compatible = "amlogic,axg-sound-card";
 		model = "X96-AIR";
@@ -120,6 +124,12 @@ &ir {
 	linux,rc-map-name = "rc-x96max";
 };
 
+&sd_emmc_a {
+	rtl8822cs: wifi@1 {
+		reg = <1>;
+	};
+};
+
 &tdmif_b {
 	status = "okay";
 };
@@ -131,3 +141,17 @@ &tdmout_b {
 &tohdmitx {
 	status = "okay";
 };
+
+&uart_A {
+	status = "okay";
+	pinctrl-0 = <&uart_a_pins>, <&uart_a_cts_rts_pins>;
+	pinctrl-names = "default";
+	uart-has-rtscts;
+
+	bluetooth {
+		compatible = "realtek,rtl8822cs-bt";
+		enable-gpios  = <&gpio GPIOX_17 GPIO_ACTIVE_HIGH>;
+		host-wake-gpios = <&gpio GPIOX_19 GPIO_ACTIVE_HIGH>;
+		device-wake-gpios = <&gpio GPIOX_18 GPIO_ACTIVE_HIGH>;
+	};
+};
-- 
2.34.1

